superset/db_engine_specs/questdb.py

```diff
@@ -0,0 +1,311 @@
+        if duration:
+            name = builtin_time_grains[duration]
+            ret_list.append(TimeGrain(name, _(name), func, duration))
+    _engine_time_grains = tuple(ret_list)
+    _default_column_type_mappings = (
+        (re.compile("^LONG256", re.IGNORECASE), types.Long256, GenericDataType.STRING),
+        (re.compile("^BOOLEAN", re.IGNORECASE), types.Boolean, GenericDataType.BOOLEAN),
+        (re.compile("^BYTE", re.IGNORECASE), types.Byte, GenericDataType.BOOLEAN),
+        (re.compile("^SHORT", re.IGNORECASE), types.Short, GenericDataType.NUMERIC),
+        (re.compile("^INT", re.IGNORECASE), types.Int, GenericDataType.NUMERIC),
+        (re.compile("^LONG", re.IGNORECASE), types.Long, GenericDataType.NUMERIC),
+        (re.compile("^FLOAT", re.IGNORECASE), types.Float, GenericDataType.NUMERIC),
+        (re.compile("^DOUBLE'", re.IGNORECASE), types.Double, GenericDataType.NUMERIC),
+        (re.compile("^SYMBOL", re.IGNORECASE), types.Symbol, GenericDataType.STRING),
+        (re.compile("^STRING", re.IGNORECASE), types.String, GenericDataType.STRING),
+        (re.compile("^UUID", re.IGNORECASE), types.UUID, GenericDataType.STRING),
+        (re.compile("^CHAR", re.IGNORECASE), types.Char, GenericDataType.STRING),
+        (re.compile("^TIMESTAMP", re.IGNORECASE), types.Timestamp, GenericDataType.TEMPORAL),
+        (re.compile("^DATE", re.IGNORECASE), types.Date, GenericDataType.TEMPORAL),
+        (re.compile(r"^GEOHASH\(\d+[b|c]\)", re.IGNORECASE), types.GeohashLong, GenericDataType.STRING)
+    )
+    column_type_mappings = _default_column_type_mappings
+
+    @classmethod
+    def build_sqlalchemy_uri(
+            cls,
+            parameters: BasicParametersType,
+            encrypted_extra: Optional[Dict[str, str]] = None
+    ) -> str:
+        return connection_uri(
+            parameters.get("host"),
+            int(parameters.get("port")),
+            parameters.get("username"),
+            parameters.get("password"),
+            parameters.get("database"))
+
+    @classmethod
+    def get_default_schema_for_query(cls, database, query) -> Optional[str]:
+        return 'public'
+
+    @classmethod
+    def get_text_clause(cls, clause):
+        """SQLAlchemy wrapper to ensure text clauses are escaped properly
+        :param clause: string clause with potentially unescaped characters
+        :return: text clause with escaped characters
+        """
+        if cls.allows_escaped_colons:
+            clause = clause.replace(":", "\\:")
+        return text(remove_public_schema(clause))
+
+    @classmethod
+    def get_time_grain_expressions(cls) -> Dict[Optional[str], str]:
+        """Return a dict of all supported time grains including any
+        potential added grains but excluding any potentially disabled
+        grains in the config file.
+        :return: All time grain expressions supported by the engine
+        """
+        return cls._time_grain_expressions
+
```

Comment:
I don't think you need to override this - it should already be handled by `BaseEngineSpec`. Or is there some reason why that method doesn't work here?

Reply:
you are correct, tahnk you for spotting all these, addressed in the same commit 9a3c588dc495ad0435666f41b455d54ee57b40fc